apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor-preview.tsx

```diff
@@ -57,7 +64,10 @@ export const InAppEditorPreview = (props: InAppEditorPreviewProps) => {
     }, 0);
 
     return () => clearTimeout(timeout);
-  }, [value]);
+  }, [editorValue]);
+
+  const previewResult = previewData?.result;
+  const preview = previewResult?.preview as InAppRenderOutput | undefined;
```

Comment:
I don't like this, but unfortunately, with the current implementation, we can't show the skeleton loader optimistically.
To have the types automatically picked we need to do check for the type: 
```
if (previewData?.result?.type !== IN_APP) {
   // RENDER SKELETON
}

// RENDER IN_APP PREVIEW
```
This requires a lot of refactoring in this file.